### DISCUSSION TITLE
Scenario "Salta": Docker container won't start

### DISCUSSION THREAD

**Discussion by fduran** [SadServers/sadservers#27] [Scenarios]
Discussions about the scenario [Salta](https://github.com/fduran/sadservers/tree/main/scenarios/salta)

**Comment by zerotrustprivacy** [comment]
I opened the ports as instructed but I'm still receiving an error message."sudo docker run -d -p 8888:8888 app node server.js
7d7f008ca546358e6d282a501c1b6cefcdf9d6725fd884b7e847366d25364f99
docker: Error response from daemon: driver failed programming external connectivity on endpoint nice_kalam (ed46d09e368bcd785e64c52d275a296bb56c731106604c6d1ef85aa8157fc6a8): Error starting userland proxy: listen tcp4 0.0.0.0:8888: bind: address already in use."

**Reply by pano9000** [reply]
probably a bit late for a reply, but maybe it's useful for others as well:

The error message is pretty "on the nose" :-) 
pay attention to the last line `0.0.0.0:8888: bind: address already in use` and combine that info with e.g. the output you see from 
`ss -tlpn`, there might be a different service running already on that port - which you might need to stop :-)

(had the same issue and though I would just use it as reverse proxy, however the Scenario's solution does NOT allow that as solution, even though `curl localhost:8888` was giving the correct output. Reason being: the check.sh is explicitly checking that you expose port 8888:8888...)

**Comment by fduran** [comment]
we've rebuilt this scenario
